src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -56,120 +55,77 @@ public sealed class RequiresAssemblyFilesAnalyzer : DiagnosticAnalyzer
 
 		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create (s_locationRule, s_getFilesRule, s_requiresAssemblyFilesRule);
 
-		public override void Initialize (AnalysisContext context)
-		{
-			context.EnableConcurrentExecution ();
-			context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.ReportDiagnostics);
+		protected override string RequiresAttributeName => RequiresAssemblyFilesAttribute;
 
-			context.RegisterCompilationStartAction (context => {
-				var compilation = context.Compilation;
+		protected override string RequiresAttributeFullyQualifiedName => RequiresAssemblyFilesAttributeFullyQualifiedName;
 
-				var isSingleFileAnalyzerEnabled = context.Options.GetMSBuildPropertyValue (MSBuildPropertyOptionNames.EnableSingleFileAnalyzer, compilation);
-				if (!string.Equals (isSingleFileAnalyzerEnabled?.Trim (), "true", StringComparison.OrdinalIgnoreCase))
-					return;
+		protected override DiagnosticTargets AnalyzerDiagnosticTargets => DiagnosticTargets.MethodOrConstructor | DiagnosticTargets.Property | DiagnosticTargets.Event;
 
-				var includesAllContent = context.Options.GetMSBuildPropertyValue (MSBuildPropertyOptionNames.IncludeAllContentForSelfExtract, compilation);
-				if (string.Equals (includesAllContent?.Trim (), "true", StringComparison.OrdinalIgnoreCase))
-					return;
+		protected override DiagnosticDescriptor RequiresDiagnosticRule => s_requiresAssemblyFilesRule;
 
-				var dangerousPatternsBuilder = ImmutableArray.CreateBuilder<ISymbol> ();
+		protected override bool VerifyMSBuildOptions (AnalyzerOptions options, Compilation compilation)
+		{
+			var isSingleFileAnalyzerEnabled = options.GetMSBuildPropertyValue (MSBuildPropertyOptionNames.EnableSingleFileAnalyzer, compilation);
+			if (!string.Equals (isSingleFileAnalyzerEnabled?.Trim (), "true", StringComparison.OrdinalIgnoreCase))
+				return true;
```

Comment:
This feels backwards. My intuition is that a function called `Verify` would return `true` on success and `false` on failure, but this seems to do the opposite.

Reply:
This doesn't seem resolved to me. Maybe call it `IsAnalyzerEnabled` to be clear on the purpose of the check?